test/integration_test/basic_test.go

```diff
@@ -308,31 +321,128 @@ func BasicCsiRegression(tc *types.TestCase) func(*testing.T) {
 		cluster, err = operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
 		require.NoError(t, err)
 		cluster.Spec.FeatureGates = map[string]string{"CSI": "false"}
-		cluster, err = ci_utils.UpdateStorageCluster(cluster)
+		ci_utils.UpdateAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
```

Comment:
can we check if the updated cluster.Spec has CSI set to false?

Reply:
Added those checks